src/test/java/io/r2dbc/proxy/support/QueryExecutionInfoFormatterTest.java

```diff
@@ -0,0 +1,546 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.r2dbc.proxy.support;
+
+import io.r2dbc.proxy.core.Bindings;
+import io.r2dbc.proxy.core.BoundValue;
+import io.r2dbc.proxy.core.ConnectionInfo;
+import io.r2dbc.proxy.core.ExecutionType;
+import io.r2dbc.proxy.core.QueryExecutionInfo;
+import io.r2dbc.proxy.core.QueryInfo;
+import org.junit.jupiter.api.Test;
+
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.Arrays;
+import java.util.Collections;
+
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author Tadaya Tsuyukubo
+ */
+public class QueryExecutionInfoFormatterTest {
+
+    @Test
+    void batchExecution() {
+
+        ConnectionInfo connectionInfo = mock(ConnectionInfo.class);
+        when(connectionInfo.getConnectionId()).thenReturn("conn-id");
+
+        // Batch Query
+        QueryExecutionInfo execInfo = mock(QueryExecutionInfo.class);
```

Comment:
This isn't going to be a binding requirement, but I'd recommend creating a `test` package with full-fledged Mock types for all (or perhaps most important) interface types.  It's pretty obvious that both users _and_ the project need to create fake versions of these types and rather than requiring users to include a mocking library, and write the behavior, wouldn't it be nicer to say:

```java
MockQueryExecutionInfo executionInfo = MockQueryExecutionInfoInfo.builder()
    .threadName("my-thread")
    .threadId(99L)
    .connectionInfo(MockConnectionInfo.builder()
        .connectionId("conn-id")
        .build())
    .success(true)
    .executeDuration(Duration.ofMillis(35))
    .type(ExecutionType.BATCH)
    .batchSize(20)
    .bindingsSize(10)
    .queries("SELECT A", "SELECT B")
    .build()